homeassistant/components/sensor/edp_redy.py

```diff
@@ -0,0 +1,122 @@
+"""Support for EDP re:dy sensors."""
+import logging
+
+from homeassistant.helpers.entity import Entity
+
+from homeassistant.components.edp_redy import EdpRedyDevice, EDP_REDY
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['edp_redy']
+
+# Load power in watts (W)
+ATTR_ACTIVE_POWER = 'active_power'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Perform the setup for re:dy devices."""
+    from edp_redy.session import ACTIVE_POWER_ID
+
+    session = hass.data[EDP_REDY]
+    devices = []
+
+    # Create sensors for modules
+    for device_json in session.modules_dict.values():
+        if "HA_POWER_METER" not in device_json["Capabilities"]:
+            continue
+        devices.append(EdpRedyModuleSensor(session, device_json))
+
+    # Create a sensor for global active power
+    devices.append(EdpRedySensor(session, ACTIVE_POWER_ID, "Power Home",
+                                 "mdi:flash", "W"))
+
+    add_devices(devices)
+
+
+class EdpRedySensor(EdpRedyDevice, Entity):
+    """Representation of a EDP re:dy generic sensor."""
+
+    def __init__(self, session, sensor_id, name, icon, unit):
+        """Initialize the sensor."""
+        EdpRedyDevice.__init__(self, session, sensor_id, name)
+
+        self._icon = icon
+        self._unit = unit
+
+        self._update_state()
```

Comment:
Remove this and call `async_add_entities` with second argument `True` instead. That will update the entity during entity addition.